### PR TITLE
msrv: serve zvol-backed patch envelope artifacts with correct size

### PR DESCRIPTION
**Pull request by andrewd-zededa**
## Description 

handlePatchFileDownload served binary artifacts via http.ServeFile, which derives Content-Length from os.Stat().Size(). For a patch envelope artifact backed by a volume, the served path is the volume's FileLocation. On ZFS that is a zvol block device whose stat size is 0, so the artifact was returned with Content-Length: 0 and an empty body, and the requesting app instance could never fetch it. File- and qcow-backed volumes were unaffected because their stat size is correct.

Add serveBinaryBlob: regular files keep using http.ServeFile, preserving Range and conditional-request support. For a non-regular backing the content length is taken from blob.Size (VolumeStatus.TotalSize, the content-tree size) and exactly that many bytes are copied. Serving exactly blob.Size also drops the trailing padding of a zvol, whose size is rounded up to the volume block size and would otherwise corrupt the artifact checksum. A non-regular backing with an unknown size now fails with 500 rather than silently serving an empty body.

Add unit tests covering the regular-file path, the block-device path (a FIFO reproduces a zvol's zero stat size and trailing padding), and the unknown-size error case.

## PR dependencies

None

## How to test and validate this PR

- deploy eve HV=kvm with zfs persist
- deploy a VM app instance (eg. ubuntu with a local network instance and route to msrv)
- attach a patch envelope with an artifact backed by a local block device volume instance
- Verify the size reported in the patch envelope artifact description.json http endpoint output is correct

## Changelog notes

Fix size reported by disk backed patch envelope artifacts from ZFS persist backed HV=kvm nodes

## PR Backports

- 17.0-stable: Yes
- 16.0-stable: Yes
- 14.5-stable: Yes
- 13.4-stable: No, as the feature is not available there.

## Checklist

- [x] I've provided a proper description
- [x] I've added the proper documentation
- [x] I've tested my PR on amd64 device
- [ ] I've tested my PR on arm64 device
- [x] I've written the test verification instructions
- [x] I've set the proper labels to this PR

And the last but not least:

- [x] I've checked the boxes above, or I've provided a good reason why I didn't
  check them.

Please, check the boxes above after submitting the PR in interactive mode.

### CODE REVIEW

File: pkg/pillar/cmd/msrv/handlers.go [hunk 1]
```diff
@@ -683,6 +683,64 @@ func (msrv *Msrv) handlePatchFileDownload() func(http.ResponseWriter, *http.Requ
...
+	}
+
+	if fi.Mode().IsRegular() {
+		http.ServeFile(w, r, blob.URL)
+		return
+	}
+
+	// Non-regular file (block device, fifo, ...): os.Stat size is unreliable,
+	// so we must know the content length up front.
+	if blob.Size <= 0 {
```
Comment (eriknordmark) [inline pkg/pillar/cmd/msrv/handlers.go:715]: I thought you said that you get Size 0, but here you generate an error if zero. Is the description wrong or I misread it?

**Review by eriknordmark — APPROVED** [review]
Run tests
